metadata.go

```diff
@@ -0,0 +1,52 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/h2oai/go-druid/builder/query"
+)
+
+type count struct {
+	Cnt int `json:"cnt"`
+}
+
+// MetadataService is a service that runs druid metadata requests using druid SQL API.
+type MetadataService struct {
+	client *Client
+}
+
+// AwaitDataSourceAvailable awaits for a datasource to be visible in druid table listing.
+func (md *MetadataService) AwaitDataSourceAvailable(dataSourceName string) error {
+	q := query.NewSQL()
+	q.SetQuery("SELECT count(*) cnt FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA = 'druid' and TABLE_NAME='" + dataSourceName + "'")
+	for range time.Tick(100 * time.Millisecond) {
```

Comment:
polling period should IMHO be configurable. 100 ms seems like hammering.

Are there alternative non-polling options how to wait for resource become available?

Reply:
Thanks! Done.